### ISSUE TITLE
King of the Oathbreakers bug

### ISSUE THREAD

**Issue by Bdett151** [magefree/mage#10634]
If you target and then phase out your spirits on another player's turn, the game freezes on the end step before your turn and will not progress to your turn.

**Comment by Susucre** [comment]
I could not reproduce this.

**Comment by Broomistermilk** [comment]
Alright so I have had this bug 3 out of 3 times in online games with players, but I couldn't reproduce it with bots. Just set it up with some people and grabbed screenshots this time. It's a Commander game with 3 players. 

Basically on my previous turn I cast Damnation, then cast an instant on King of the Oathbreakers to phase him out. My opponents' turns go by and the glitch begins on the player before mine's endstep as you can see in the first screenshot (my opponents POV is the second screenshot) . I tried the reconnect fix, launching from the launcher again, which lead to the third screenshot. As you can see it's now my upkeep and King of the Oathbreakers is on the board (without the extra spirit) and neither me or my opponents have anything to click. Then tried /fix twice which are the fourth and fifth screenshots. The player before me also tried the reconnect fix to no avail.

I hope this helps! 

![King of the Oathbreakers Bug A](https://github.com/magefree/mage/assets/139816297/6e10d1d1-68e7-4893-ba7d-855745f1c798)
![King of the Oathbreakers Bug A-1](https://github.com/magefree/mage/assets/139816297/e98f17ad-851d-408a-822e-6f2cd82c342e)
![King of the Oathbreakers Bug B](https://github.com/magefree/mage/assets/139816297/c0458baf-45f7-4c6e-9270-19ced82f1ef1)
![King of the Oathbreakers Bug C](https://github.com/magefree/mage/assets/139816297/4fe9bc8d-d62c-4e7a-8d2f-de301ac289ba)
![King of the Oathbreakers Bug D](https://github.com/magefree/mage/assets/139816297/6d110b11-ec61-46f4-bfac-38cc9d86a66d)

**Comment by Susucre** [comment]
Thanks for the additional info.
I did end up replicating the bug having 2 human clients. In the same game I did target the King with multiple bolts, and the third time it phased in, the game was locked.

I had the following error in the server logs, repeated multiple time, and again on attempting to reconnect with a fresh client:
> WARN  2023-07-27 01:12:10,780 Game frozen in waitResponseOpen for user Susucr (connection problem)                        =>[CALL main-86] HumanPlayer.waitResponseOpen

So my guess is something is bugged on phasing in permanents at upkeep, and triggering from there. It seems to not always happen however (first 2 phasing in worked), so maybe that is an asynchronous issue?
